contracts/aave/AaveV2Contract.sol

```diff
@@ -0,0 +1,173 @@
+
+    //----------------------------//
+    //          Events            //
+    //----------------------------//
+
+    event AddAsset(address indexed _asset, address _oracle);
+    event BorrowRatioChanged(uint256 indexed ratio);
+
+    //----------------------------//
+    //          Modifiers         //
+    //----------------------------//
+
+    modifier isValid(address _asset) {
+        require(depositAssets[_asset] == address(0));
+        _;
+    }
+
+    //----------------------------//
+    //       View Functions       //
+    //----------------------------//
+
+    /**
+     * @dev fetch the latest price of the asset from oracle
+     * @param _asset address for which we need price
+     */
+    function getLatestPrice(address _asset) public view returns (int256 price) {
+        AggregatorV3Interface priceFeed = AggregatorV3Interface(_asset);
+        (, price, , , ) = priceFeed.latestRoundData();
+    }
+
+    function nomalize(uint256 amount) public view returns (uint256) {
+        return amount * 10**(ETH_DECIMALS - USD_DECIMALS);
+    }
+
+    //----------------------------//
+    //     Mutation Functions     //
+    //----------------------------//
+
+    /**
+     * @dev Only owner can add new assets that the contract will support for staking
+     * @param _asset will be address of the asset
+     * @param _priceOracle will the chainlink asset price oracle
+     * @return event AddAsset(_asset: 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2, _priceOracle: 0xAc559F25B1619171CbC396a50854A3240b6A4e99)
+     */
+    function addDepositAsset(
+        address _asset,
+        address _priceOracle
+    ) public onlyOwner isValid( _asset) {
+        depositAssets[_asset] = _priceOracle;
+        emit AddAsset(_asset,_priceOracle);
+    }
+
+    /**
+     * @dev Users will deposit their assets in pool
+     * @param _asset will be asset name like ETH or BTC
+     * @param _amount will be the deposited amount
+     */
+    function deposit( address _asset,uint256 _amount) public nonReentrant payable {
+        require(
+            depositAssets[_asset] != address(0),
+            Errors.ASSET_NOT_ALLOWED
+        );
+
+        int256 assetPrice = getLatestPrice(depositAssets[_asset]);
+
+        if (msg.value > 0) {
+            WETHGATEWAY.depositETH{value: msg.value}(
+                address(POOL),
+                address(this),
+                0
+            );
+        } else {
+            require(_amount > 0, Errors.ZERO_AMOUNT);
+            IERC20 supplyAsset = IERC20(_asset);
+            supplyAsset.transferFrom(msg.sender, address(this), _amount);
+            supplyAsset.approve(address(POOL), _amount);
+            POOL.deposit(address(supplyAsset), _amount, address(this), 0);
+        }
+    }
+
+    /**
+     * @dev Cruize will borrow on behalf of user
+     * @param _asset will be asset name like ETH or BTC
+     */
+    function borrow(address _asset) public nonReentrant {
```

Comment:
borrow its a public function 😂   why just think guys anyone would be able to borrow lone I don't see any check for this what's going on